modules/lead/product/dockercfg.tf

```diff
@@ -0,0 +1,28 @@
+data "template_file" "dockercfg" {
+  template = file("${path.module}/dockercfg.tpl")
+ 
+  vars = { 
+    email = "jenkins@liatr.io"
+    artifactory_url   = "https://artifactory.toolchain.${var.cluster_domain}/docker-registry/${var.product_name}"
+    artifactory_auth = base64encode(
+      "${data.kubernetes_secret.jenkins_artifactory_credential.data.username}:${data.kubernetes_secret.jenkins_artifactory_credential.data.password}",
+    )   
+    harbor_url = "https://harbor.toolchain.${var.cluster_domain}/${var.product_name}"
+    harbor_auth = base64encode(
+      "robot$imagepusher:${data.kubernetes_secret.product-harbor-creds.data.AUTH}",
```

Comment:
Did the idea change of having multiple robot accounts specific for each product or repo?

Reply:
No, harbor projects and auth resources are still created per product. The dockercfg contains both artifactory and harbor credentials so pipelines can push to either

Reply:
hmm maybe i'm just not understanding the robot fully. I was assuming where you have `imagepusher` you would have to have the name of the specific robot account that gets created for that repo.

Reply:
Every product has a robot account named `robot$imagepusher`, which is tied to the product by the specific auth token (robot$imagepusher:AUTHTOKEN). The token for the jordan-lab `robot$imagepusher` robot account won't work for the jon-lab `robot$imagepusher` and vice versa.

Reply:
thanks for the explanation